File: leetcode/77. Combinations/main.go
package main

func combine(n int, k int) [][]int {
	ans := [][]int{}

	var dfs func([]int, int)
	dfs = func(path []int, prev int) {
		if len(path) == k {
			temp := make([]int, k)
			copy(temp, path)
			ans = append(ans, temp)
		}

		for i := prev; i <= n; i++ {
			path = append(path, i)
			dfs(path, i+1)
			path = path[:len(path)-1]
		}
	}

	dfs([]int{}, 1)

	return ans
}

File: leetcode/77. Combinations/main.ts
const combine = (n: number, k: number): number[][] => {
  let nums = [...Array(n).keys()].map((val) => val + 1)
  let ans: number[][] = []

  const dfs = (cur: number[], i: number) => {
    if (cur.length == k) {
      ans.push([...cur])
      return
    } else {
      for (let j = i; j < n; j++) {
        cur.push(nums[j])
        dfs(cur, j + 1)
        cur.pop()
      }
    }
  }

  dfs([], 0)

  return ans
}
